ztp/siteconfig-generator/siteConfig/siteConfigHelper.go

```diff
@@ -273,3 +336,47 @@ func mergeJsonCommonKey(mergeWith, mergeTo, key string) (string, error) {
 	}
 	return string(newJson), nil
 }
+
+func applyWorkloadPinningInstallConfigOverrides(clusterSpec *Clusters) (result string, err error) {
+	const (
+		deprecationCR      = "AgentClusterInstall"
+		deprecationField   = "cpuset"
+		deprecationMessage = "siteConfig.clusters[].node[].cpuset will be deprecated after OCP 4.15, please use siteConfig.clusters[].cpuPartitioningMode for OCP versions >= 4.13"
+	)
+
+	nodeHasCPUSet := false
+	for _, node := range clusterSpec.Nodes {
+		// If a CPUSet exists on any of the nodes, the whole cluster, regardless if SNO or not, must be setup
+		// for partitioning.
+		if node.Cpuset != "" {
+			nodeHasCPUSet = true
+			break
+		}
+	}
+
+	if clusterSpec.CPUPartitioning == CPUPartitioningAllNodes {
+		defaultValues := map[string]interface{}{
+			"cpuPartitioningMode": "AllNodes",
+		}
+		if clusterSpec.InstallConfigOverrides != "" {
+			err := json.Unmarshal([]byte(clusterSpec.InstallConfigOverrides), &defaultValues)
```

Comment:
Not sure how we feel about this behavior, if someone supplies `installConfigOverrides` in the siteConfig, do they take higher priority, or should we assume the flag is of `cpuPartitioningMode` takes higher priority and always alters what ever the value might be in the `installConfigOverrides`?

Reply:
There is precedent in how the networkType field is handled in which case the explicit field overwrites content in the installConfigOverrides. See this comment thread for the discussion: https://github.com/openshift-kni/cnf-features-deploy/pull/1316#discussion_r1019461262

Reply:
That's good to know, let me update that.